プルリクエスト：体力テスト結果ページ作成ソース

hkaneko16 opened the pull request:
体力テスト結果ページ作成ソースのプルリクエスト

hkaneko16 committed:
コミット:体力テスト結果ページ作成ソース

ma0125 commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	 * @param categoryValue 値
+	 * @return td要素
+	 */
+	private String createTd(String categoryValue) { // TODO: otsuka: なんか至る所で「physicalTestResult」って変数名出てくるけど、これって何者？JSONObjectだったり、Stringだったり、すごい混乱します。しかも結果って名前がついているのに、「user_name」「blood_type」とかも渡ってくるっぽい。。。 kaneko: physicalTestResultから「項目の値」という意味でcategoryValueに変更しました。
```
引数のcategoryValueはvalueでいいと思います！

hkaneko16 replied:
categoryValueからvalueに変更しました。

ma0125 commented on PhysicalTestResult/src/study/html/HtmlPrinter.java:
```diff
@@ -0,0 +1,27 @@
+	/**
+	 * HTMLファイルを生成する。
+	 *
+	 * @param printString ファイルに書き込む文字列 TODO: otsuka: 引数名とコメントあってないよね。 kaneko: 引数名をhtmlから変更しました。
```
HTMLファイルを生成するメソッドなので、引数はhtmlでいいかも。
HTMLを書き込んでHTMLファイルを作成しているので・・・

otameshi-developer replied:
不要なコメントは削除しましょう。

-----
 TODO: otsuka: 引数名とコメントあってないよね。 kaneko: 引数名をhtmlから変更しました。

otameshi-developer commented on PhysicalTestResult/.classpath:
```diff
@@ -0,0 +1,7 @@
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
+	<classpathentry kind="lib" path="C:/Users/Hiroto/Downloads/json-20190722.jar"/>
```
こういう個人名が入ったパスとかも上げない方がいいと思います。
Cドラ直下にフォルダ作成してそこに格納するとか工夫しましょう。

hkaneko16 replied:
Cドラ直下にDownloadsフォルダを作成して、そこに格納しました。

otameshi-developer replied:
新規コメント記載しました。
※僕もまだGitHub不慣れなところがあるので、ちょっと違う会話として登録されてしまった。。ので、スレッド後半の方を見てみてください。
ひとまず、この会話はresolvedとします。

ma0125 commented on PhysicalTestResult/src/study/json/ListSorter.java:
```diff
@@ -0,0 +1,39 @@
+	/**
+	 * リストの中身をソートする。
+	 * 
+	 * @param physicalTestResult 体力テストの結果リスト // TODO: otsuka: 引数名違う。あと、ここは「体力テストの結果」なのに、returnで使っている文言は「体力テストの結果リスト」と、統一性がない。 kaneko: 引数名を訂正し、returnの文言と統一しました。
```
このphysicalTestResultのコメントはリストではなくJSONでは？
メソッドのコメントもJSONをリストに入れてソートしているという感じがいいと思います！

ma0125 commented on PhysicalTestResult/src/study/json/PhysicalTestResult.java:
```diff
@@ -0,0 +1,96 @@
+		 * TODO: otsuka: compareNumberとcomparedNumber
+		 * 比較する値と比較された値？ってことかな？意味あってる？ kaneko: 比較する値と比較される値という意味で付けていました。比較元の値と比較する値という意味に変更します。
+		 */
+		int comparisonSourceNumber = this.testResult.getInt(KEY_NAME_STUDENT_NUMBER);
```
この変数名はbaseとかでもいいと思います！

hkaneko16 replied:
comparisonSourceNumberをbaseに、compareNumberを比較対象という意味でcomparisonに変更しました。

otameshi-developer replied:
ここのok、ngは @ma0125 に任せます！笑

こういう命名ってこれからも苦労すると思うけど、頑張ってｗ
このパターンで個人的によく使うのは↓↓この辺のキーワードかなぁ。
「my」「own」「other」「target」「base」

特に今回のパターンだと、自分自身（this）と引数で渡ってきた自分以外を比較するわけだから
my vs otherとかown vs otherとかでもまぁまぁしっくりくると思っている。

otameshi-developer commented on PhysicalTestResult/res/base.html:
```diff
@@ -0,0 +1,26 @@
+                    名前
+                </th>
+                <th>
+                    上体起こし(回)<!-- TODO: otsuka: 素朴な疑問。体力テストの結果に血液型って関係あるの？？ kaneko: 体力テストという観点から外れていたので上体起こしに変更しました。 -->
```
ソースとして不要なコメントはどんどん削除していこうか。
履歴としてコメントの内容は下記に貼り付けておきます。

-----
TODO: otsuka: 素朴な疑問。体力テストの結果に血液型って関係あるの？？ kaneko: 体力テストという観点から外れていたので上体起こしに変更しました。

otameshi-developer replied:
<!-- TODO: otsuka: 素朴な疑問。体力テストの結果に血液型って関係あるの？？ kaneko: 体力テストという観点から外れていたので上体起こしに変更しました。 -->

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer replied:
確認しました。okです。

otameshi-developer commented on PhysicalTestResult/res/physicalTestResult.html:
```diff
@@ -0,0 +1,44 @@
+<html>
```
このファイルって最終的な生成物？？
だとしたら、resフォルダ配下にあるのはおかしいよね？

otameshi-developer commented on PhysicalTestResult/src/study/main/Main.java:
```diff
@@ -0,0 +1,60 @@
+import study.json.PhysicalTestResult;
+
+/**
+ * TODO: otsuka: Project全体的な指摘はここに記載します。
+ * ◆空行の先頭にタブがある箇所がいくつかあります。
+ * 　一般的なサーバーとかだったらパワーあるからあんまり気にしなくていいところかもしれないけど、
+ * 　MEAPとかだと出来るだけリソースの消費を抑えたいので、削除する癖をつけてください。
+ * 　この時、eclipseで、自動で末尾のスペースやタブをトリムする機能があるけど
+ * 　これは使用しないでください。
+ * 　特に既存アプリのソースに対して、勝手に空白削除とかすると差分がえげつない量でてしまうので。
+ *
+ * ◆空行の開け方がバラバラ。
+ * 　個人的には「こーゆーところで改行しろ！」みたいな強制はしないけど、
+ * 　自身の中では統一性を持っていてほしい。
+ * 　※※大前提として、既存ソースは既存のルールに合わせること。自分色を主張するとソースが読みにくくなる。※※
+ * 　メソッドの開始終了は1行空ける。とか、何かしら自分のルールを持って、どう書いたら読みやすいかな？ってのを
+ * 　常に考えてください。
+ *
+ * ◆base.htmlとphysicalTestData.jsonの置き場所
+ * 　なんでsrcフォルダ内？resじゃない？ kaneko: MEAPの取り方でリソースを取得していたのですが、その場合srcフォルダ内でないと取得できなかったので、srcフォルダ内に置いていました。Fileインスタンスを生成する方法だとresから取得できたので変更します。
```
不要なコメントは削除しましょう。
ちなみに、
「MEAPの取り方でリソースを取得していたのですが、その場合srcフォルダ内でないと取得できなかったので」これって具体的に、なんで出来なかったの？


-----
 * TODO: otsuka: Project全体的な指摘はここに記載します。
 * ◆空行の先頭にタブがある箇所がいくつかあります。
 * 　一般的なサーバーとかだったらパワーあるからあんまり気にしなくていいところかもしれないけど、
 * 　MEAPとかだと出来るだけリソースの消費を抑えたいので、削除する癖をつけてください。
 * 　この時、eclipseで、自動で末尾のスペースやタブをトリムする機能があるけど
 * 　これは使用しないでください。
 * 　特に既存アプリのソースに対して、勝手に空白削除とかすると差分がえげつない量でてしまうので。
 *
 * ◆空行の開け方がバラバラ。
 * 　個人的には「こーゆーところで改行しろ！」みたいな強制はしないけど、
 * 　自身の中では統一性を持っていてほしい。
 * 　※※大前提として、既存ソースは既存のルールに合わせること。自分色を主張するとソースが読みにくくなる。※※
 * 　メソッドの開始終了は1行空ける。とか、何かしら自分のルールを持って、どう書いたら読みやすいかな？ってのを
 * 　常に考えてください。
 *
 * ◆base.htmlとphysicalTestData.jsonの置き場所
 * 　なんでsrcフォルダ内？resじゃない？ kaneko: MEAPの取り方でリソースを取得していたのですが、その場合srcフォルダ内でないと取得できなかったので、srcフォルダ内に置いていました。Fileインスタンスを生成する方法だとresから取得できたので変更します。

hkaneko16 replied:
不要コメント、削除しました。
また、リソースを取得できなかった理由は、resフォルダをクラスパスに登録していなかったことが原因でした。
そのため、NullPointerExceptionが発生してしまったのですが、
eclipse上での設定から登録することでリソースの取得が可能になりました。

otameshi-developer replied:
ok!
「単純に変更した」ってだけじゃなくて、疑問点などは解消しながら前に進んでいきましょう。

otameshi-developer commented on PhysicalTestResult/src/study/main/Main.java:
```diff
@@ -0,0 +1,60 @@
+ * 体力テストの結果を使ってHTMLファイルを生成する。
+ *
+ */
+public class Main { // TODO: otsuka:Mainってjsonパッケージの中がいいの？ kaneko:Mainのパッケージは考えていなかったです。新たにmainパッケージを作って、その中にしました。
```
不要なコメントは削除しましょう。

-----
// TODO: otsuka:Mainってjsonパッケージの中がいいの？ kaneko:Mainのパッケージは考えていなかったです。新たにmainパッケージを作って、その中にしました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer replied:
ok!

otameshi-developer commented on PhysicalTestResult/src/study/main/Main.java:
```diff
@@ -0,0 +1,60 @@
+			String html = new HtmlBuilder().createHtml(results);
+			new HtmlPrinter().printHtml(html);
+
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (JSONException e) {
+			e.printStackTrace();
+		}
```
この2つのcatchは同じことしているんだけど、分ける意味あるのかな？

hkaneko16 replied:
同じcatch句で複数の例外をcatchするように変更しました。

otameshi-developer replied:
ok!
ただ、ここのJSONObjectは不要です。
いまはこのままで良いけど、他の箇所を修正する際に消してもらうと思います。

ひとまず、コメント読んだらここはresolvedにしちゃってください。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+ * 体力テスト結果のHTMLを生成する。
+ *
+ */
+public class HtmlBuilder {
```
せっかく各メソッドをcreateに変えたなら、クラス名も変わるんじゃない？

hkaneko16 replied:
クラス名をHtmlBuilderからHtmlCreatorに変更しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+
+	private static final String REPLACE_MARK = "$$";
+
+	// TODO: otsuka: なんでHTML担当の人がJSONのキー知ってるの？？役割が違うんじゃない？あと、KEY_NAMEのNAMEっているかな？もしくはKEY kaneko: JSONのキーはPhysicalTestResultに持たせました。
+//	private static final String KEY_NAME_USER_NAME = "user_name";
+
+//	private static final String KEY_NAME_BLOOD_TYPE = "blood_type";
+
+//	private static final String KEY_NAME_50M_RUN = "50m_run"; // TODO: otsuka: 数字始まりは微妙。一応使えるけど、、、JSONってJavaScriptをベースにしてるからなぁ（JSとJSONは別規格となっているので考えなくても良いかもだけど。）。。。JSは数字始まりはngのはず。参考程度だけど、Google内ではこんな感じでルールがあるみたいね[https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format]。
+
+//	private static final String KEY_NAME_GRIP = "grip";
```
不要なコメントは削除しましょう。

-----
	// TODO: otsuka: なんでHTML担当の人がJSONのキー知ってるの？？役割が違うんじゃない？あと、KEY_NAMEのNAMEっているかな？もしくはKEY kaneko: JSONのキーはPhysicalTestResultに持たせました。
//	private static final String KEY_NAME_USER_NAME = "user_name";

//	private static final String KEY_NAME_BLOOD_TYPE = "blood_type";

//	private static final String KEY_NAME_50M_RUN = "50m_run"; // TODO: otsuka: 数字始まりは微妙。一応使えるけど、、、JSONってJavaScriptをベースにしてるからなぁ（JSとJSONは別規格となっているので考えなくても良いかもだけど。）。。。JSは数字始まりはngのはず。参考程度だけど、Google内ではこんな感じでルールがあるみたいね[https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format]。

//	private static final String KEY_NAME_GRIP = "grip";

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	 * @return 体力テスト結果のHTML
+	 * @throws IOException 読み込みに失敗
+	 */
+	public String createHtml (List<PhysicalTestResult> results) throws IOException, JSONException { // TODO: kaneko: buildHtmlからcreateHtmlに変更しました。
```
不要なコメントは削除しましょう。

-----
 // TODO: kaneko: buildHtmlからcreateHtmlに変更しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	 * @param results 体力テストの結果のリスト
+	 * @return HTML形式の体力テストの結果
+	 */
+	private String createResultData(List<PhysicalTestResult> results) throws JSONException { // TODO: kaneko: buildResultDataからcreateResultDataに変更しました。
```
不要なコメントは削除しましょう。

-----
 // TODO: kaneko: buildResultDataからcreateResultDataに変更しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	 */
+	private String createResultData(List<PhysicalTestResult> results) throws JSONException { // TODO: kaneko: buildResultDataからcreateResultDataに変更しました。
+
+		StringBuilder resultList = new StringBuilder();
```
この変数名、メソッド名に合わせた方がしっくりくるね。

hkaneko16 replied:
変数名をcreateResultDataというメソッド名に合わせるために
resultListからresultDataに変更しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	private String createResultData(List<PhysicalTestResult> results) throws JSONException { // TODO: kaneko: buildResultDataからcreateResultDataに変更しました。
+
+		StringBuilder resultList = new StringBuilder();
+		Iterator<PhysicalTestResult> ite = results.iterator(); // TODO: kaneko: for文を止め、Iteratorを使ってループ処理を行うようにしました。
```
不要なコメントは削除しましょう。
あと、Iteratorにするんじゃないんじゃないかな。

-----
// TODO: kaneko: for文を止め、Iteratorを使ってループ処理を行うようにしました。

hkaneko16 replied:
不要コメント、削除しました。
また、IteratorではなくforEachを使ってループ処理を行うように変更しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	/**
+	 * tr要素を生成する。
+	 *
+	 * @param onePersonRecord １人分の記録 // TODO: kaneko: physicalTestResultから変更しました。
```
不要なコメントは削除しましょう。
あと、これってレコードなの？
PhysicalTestResult の説明にはrecordなんて登場しないけど。

-----
 // TODO: kaneko: physicalTestResultから変更しました。

hkaneko16 replied:
不要コメント、削除しました。
また、変数名をonePersonRecordからonePersonDataに変更しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	 * @param onePersonRecord １人分の記録 // TODO: kaneko: physicalTestResultから変更しました。
+	 * @return tr要素
+	 */
+	private String createTr(PhysicalTestResult onePersonRecord) throws JSONException { // TODO: kaneko: buildTrからcreateTrに変更しました。また、JSONObjectではなく自作クラスのPhysicalTestResultを受け取るようにしました。
```
不要なコメントは削除しましょう。

-----
 // TODO: kaneko: buildTrからcreateTrに変更しました。また、JSONObjectではなく自作クラスのPhysicalTestResultを受け取るようにしました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	private String createTr(PhysicalTestResult onePersonRecord) throws JSONException { // TODO: kaneko: buildTrからcreateTrに変更しました。また、JSONObjectではなく自作クラスのPhysicalTestResultを受け取るようにしました。
+
+		StringBuilder tr = new StringBuilder();
+		tr.append("<tr>").append(CRLF); // TODO: otsuka: あんまりこれっと言って意味を持たないCRLFを1行で書かなくて良いかな（<tr>のappendの後ろに付けた方がいいんじゃない？）。例えばHTTPのパケットのヘッダーとボディの区切りとか、特別な意味を持っているなら1行で書かないと埋もれちゃうけど、これってHTMLのソース上の見た目だけのものだよね（ブラウザ上では見えないもの）？であればそんなに重要ではないと思う。 kaneko: CRLFを1行で書かずに繋げました。
```
不要なコメントは削除しましょう。

-----
// TODO: otsuka: あんまりこれっと言って意味を持たないCRLFを1行で書かなくて良いかな（<tr>のappendの後ろに付けた方がいいんじゃない？）。例えばHTTPのパケットのヘッダーとボディの区切りとか、特別な意味を持っているなら1行で書かないと埋もれちゃうけど、これってHTMLのソース上の見た目だけのものだよね（ブラウザ上では見えないもの）？であればそんなに重要ではないと思う。 kaneko: CRLFを1行で書かずに繋げました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	 * @param categoryValue 値
+	 * @return td要素
+	 */
+	private String createTd(String categoryValue) { // TODO: otsuka: なんか至る所で「physicalTestResult」って変数名出てくるけど、これって何者？JSONObjectだったり、Stringだったり、すごい混乱します。しかも結果って名前がついているのに、「user_name」「blood_type」とかも渡ってくるっぽい。。。 kaneko: physicalTestResultから「項目の値」という意味でcategoryValueに変更しました。
```
不要なコメントは削除しましょう。

-----
/ TODO: otsuka: なんか至る所で「physicalTestResult」って変数名出てくるけど、これって何者？JSONObjectだったり、Stringだったり、すごい混乱します。しかも結果って名前がついているのに、「user_name」「blood_type」とかも渡ってくるっぽい。。。 kaneko: physicalTestResultから「項目の値」という意味でcategoryValueに変更しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	}
+
+	/**
+	 * 行の中に置換文字列があった場合、マップの値を置換する。 // TODO: otsuka: これって、1行まるまる置換するメソッドだよね？行中に置換文字列が紛れ込んでたら、全体差し替えちゃう？ kaneko: 全体差し替えてしまう実装だったので、行中の置換文字列のみを差し替える実装に変更しました。
```
不要なコメントは削除しましょう。

-----
 // TODO: otsuka: これって、1行まるまる置換するメソッドだよね？行中に置換文字列が紛れ込んでたら、全体差し替えちゃう？ kaneko: 全体差し替えてしまう実装だったので、行中の置換文字列のみを差し替える実装に変更しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlBuilder.java:
```diff
@@ -0,0 +1,139 @@
+	 */
+	private String replaceLine(String line, Map<String, String> mapReplacement) {
+
+		String newLine = line; // TODO: kaneko: replacedLineからnewLineに変更しました。
```
不要なコメントは削除しましょう。

-----
 // TODO: kaneko: replacedLineからnewLineに変更しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/html/HtmlPrinter.java:
```diff
@@ -0,0 +1,27 @@
+
+		try (PrintWriter pw = new PrintWriter("res/physicalTestResult.html")) {
+			pw.print(printString);
+			pw.flush();// TODO: otsuka: Exception発生時もちゃんとcloseされる？ kaneko: try-with-resources文を使ってリソースを自動でcloseするようにしました。
```
不要なコメントは削除しましょう。

-----
			pw.flush();// TODO: otsuka: Exception発生時もちゃんとcloseされる？ kaneko: try-with-resources文を使ってリソースを自動でcloseするようにしました。
		}// TODO: otsuka: このキャッチいる？ kaneko: 上に投げるだけだったので意味なかったです。catch句は消しました。

otameshi-developer commented on PhysicalTestResult/src/study/json/FileReader.java:
```diff
@@ -0,0 +1,38 @@
+ * ファイルを読み込み、文字列を生成する。
+ *
+ */
+public class FileReader { // TODO: otsuka: あんまりこういうクラス名ってみないよね。pure javaのjavadocとか参考にしてほしいんだけど、この場合はFileReaderじゃない？（読む人はreadじゃなくてreaderじゃない？） kaneko: 確かにFileReaderです。FileReadから変更しました。
```
不要なコメントは削除しましょう。

-----
public class FileReader { // TODO: otsuka: あんまりこういうクラス名ってみないよね。pure javaのjavadocとか参考にしてほしいんだけど、この場合はFileReaderじゃない？（読む人はreadじゃなくてreaderじゃない？） kaneko: 確かにFileReaderです。FileReadから変更しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/json/FileReader.java:
```diff
@@ -0,0 +1,38 @@
+	 * ファイルを読み込み、文字列を生成する。
+	 *
+	 * @param filePath ファイルのパス
+	 * @return ファイルから生成した文字列 // TODO: otsuka: なにか変換したの？　kaneko: 変換はしていないです。「ファイルから生成した文字列」に変更します。
```
不要なコメントは削除しましょう。
あと、これはもっとストレートに表現した方がいいよ。
生成したんじゃなくて、読み込んだんでしょ？

-----
	 * @return ファイルから生成した文字列 // TODO: otsuka: なにか変換したの？　kaneko: 変換はしていないです。「ファイルから生成した文字列」に変更します。

hkaneko16 replied:
不要コメント、削除しました。
また、「ファイルから読み込んだ文字列」に変更しました。

otameshi-developer commented on PhysicalTestResult/src/study/json/FileReader.java:
```diff
@@ -0,0 +1,38 @@
+	public String read(String filePath) throws IOException {
+
+		StringBuilder fileData = new StringBuilder();
+		// TODO: otsuka: main()で実行したときって、リソースこうやって取らないといけないんだっけ？なんか普通にFileインスタンス生成すればよいような気もするんだけど。。。（MEAPはこの取り方が正解！） kaneko: Fileインスタンス生成で出来たので修正しました。
```
不要なコメントは削除しましょう。

-----
		// TODO: otsuka: main()で実行したときって、リソースこうやって取らないといけないんだっけ？なんか普通にFileインスタンス生成すればよいような気もするんだけど。。。（MEAPはこの取り方が正解！） kaneko: Fileインスタンス生成で出来たので修正しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/json/FileReader.java:
```diff
@@ -0,0 +1,38 @@
+				fileData.append(line);
+				line = br.readLine();
+			}
+		}// TODO: otsuka: これってなんでキャッチしているの？意味ある？ kaneko: 上に投げるだけだったので意味なかったです。catch句は消しました。
```
不要なコメントは削除しましょう。

-----
		}// TODO: otsuka: これってなんでキャッチしているの？意味ある？ kaneko: 上に投げるだけだったので意味なかったです。catch句は消しました。

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer commented on PhysicalTestResult/src/study/json/FileReader.java:
```diff
@@ -0,0 +1,38 @@
+		return fileData.toString();
+
+	}
+
```
//  良い機会だから、ちょっと違う方法で読み込んでみようか。
	// 実質やっていることは違うんだけど、使うAPIが違うって感じ。
	// ポイントとしては、
	// - new InputStream()とか使わない。Filesを利用してください。
	// - lines
	// ざっくりとしたポイントしか書かないようにしたので、わかんなかったら聞いてください。（まずは色々調べてみてー）

hkaneko16 replied:
Filesクラスのlinesを使用して、ファイルの全行をStreamで返却し、
forEachでappendを繰り返すように変更しました。

otameshi-developer replied:
初手としては良いのではないでしょうか！
ただ、この読み取り方ってbufferedしているのかな？
この規模ならbufferedしていなくてもokだけど、ファイルの容量が増える場合を考えると
bufferedして読み取りたいです。

hkaneko16 replied:
bufferedしていませんでした。
FilesクラスのnewBufferedReaderでBufferedReaderを返し、
BufferedReaderから読み込まれる行が要素となるStreamを返すように変更しました。

otameshi-developer replied:
> bufferedしていませんでした。
調べたの？

otameshi-developer replied:
@hkaneko16 

> bufferedしていませんでした。

調べたの？

hkaneko16 replied:
はい、JavadocのFilesクラスのlinesの説明を読んで、特にbufferedされるといった記述はなかったため、
bufferedしていないとコメントしました。

otameshi-developer commented on PhysicalTestResult/src/study/json/ListSorter.java:
```diff
@@ -0,0 +1,39 @@
+	 * @return ソート後の体力テストの結果リスト
+	 */
+	public List<PhysicalTestResult> sortList(JSONObject physicalTestResult) throws JSONException { // TODO: otsuka: Mainクラスが持っているのが気持ち悪い。他の機能はちゃんと他のクラスに分けているのに。。。
+		/*
+		 * TODO: otsuka: resultsって名前は良くないかな。
+		 * 特にこのアプリは「体力テストの結果」って意味でresultをすでに使用しているから
+		 * 紛らわしい。
+		 * ここで使うべきは、「ソート済み」ってことを伝えるキーワードじゃない？　kaneko:　「ソート済みのリスト」という意味でsortedListにしました。 
+		 */
```
不要なコメントは削除しましょう。

-----
		/*
		 * TODO: otsuka: resultsって名前は良くないかな。
		 * 特にこのアプリは「体力テストの結果」って意味でresultをすでに使用しているから
		 * 紛らわしい。
		 * ここで使うべきは、「ソート済み」ってことを伝えるキーワードじゃない？　kaneko:　「ソート済みのリスト」という意味でsortedListにしました。 
		 */

otameshi-developer commented on PhysicalTestResult/src/study/json/ListSorter.java:
```diff
@@ -0,0 +1,39 @@
+		List<PhysicalTestResult> sortedList = new ArrayList<>();
+		String[] userID = JSONObject.getNames(physicalTestResult);
+		for (int i = 0; i < userID.length; i++) {
+			sortedList.add(new PhysicalTestResult(physicalTestResult.getJSONObject(userID[i]))); // TODO: otsuka: getJSONObject()ってthrows宣言あるけど、特にエラーハンドリングしていないけど、これって発生したらどうなるの？ kaneko: 考えていなかったです。動作としてはJSONExceptionが発生してスタックトレースを吐いてプログラムは終了します。ちゃんとMainでcatchするようにしました。
```
不要なコメントは削除しましょう。
さて、Mainでキャッチするようにしたのは置いておいて。。。そもそも論でJSONExceptionは上位に投げないでください。詳しくはPhysicalTestResult.getString()に書いてあるので、そちら参照よろしく。

-----
			sortedList.add(new PhysicalTestResult(physicalTestResult.getJSONObject(userID[i]))); // TODO: otsuka: getJSONObject()ってthrows宣言あるけど、特にエラーハンドリングしていないけど、これって発生したらどうなるの？ kaneko: 考えていなかったです。動作としてはJSONExceptionが発生してスタックトレースを吐いてプログラムは終了します。ちゃんとMainでcatchするようにしました。

otameshi-developer commented on PhysicalTestResult/src/study/json/PhysicalTestResult.java:
```diff
@@ -0,0 +1,96 @@
+package study.json;
```
もろもろ不要なコメントは削除しましょう。

-----
/**
 * 体力テストの結果を保持するクラス
 *　TODO: otsuka: このクラスの説明は嫌です。
 * インターフェースって1クラスに何個でも指定できるけど、全部書くの？
 * Aインターフェースと、Bインターフェースと、Cインターフェース・・・・・・を実装したクラスです。って感じ？涙
 * ※たしか数に制限はないはず。。。
 * kaneko: インターフェースを実装した旨のコメントは消しました。
 *
 */

	// TODO: kaneko: JSONのキーはPhysicalTestResultに持たせました。また、KEY_NAMEのNAMEを消しました。

	public static final String KEY_50M_RUN = "run_50m";// TODO: kaneko: 文字から始まるように50m_runから変更しました。
	private JSONObject testResult; // TODO: otsuka: publicはダメです。外から触り放題。仮にpublicだとしたら、後述のgetterは不要です。 kaneko: privateに変更しました。

	public PhysicalTestResult(JSONObject testResult) { // TODO: otsuka: joって略す人みたことないな。。。testResultじゃダメなの？ kaneko: testResultの方が分かり易いので変更しました。

		return Integer.compare(comparisonSourceNumber, compareNumber); // TODO: commaの後ろは半角スペース空けてね

hkaneko16 replied:
不要コメント、削除しました。

otameshi-developer replied:
OK!!

otameshi-developer commented on PhysicalTestResult/src/study/json/PhysicalTestResult.java:
```diff
@@ -0,0 +1,96 @@
+	 * 
+	 * @return　体力テストの結果を保持するJSONObject
+	 */
+//	public JSONObject getJSONObject() { // TODO: これっているのかなぁ。。。いちいちjson取得しないといけない？ kaneko: JSONObjectを取得することを止め、getStringとgetIntを用意しました。
+//		return this.testResult;
+//	}
```
いい感じだけど、もう一声！
このクラスからメンバのtestResultを取っ払ってください！
※以前ソースに書き込んだコメントから変更しています。

hkaneko16 replied:
JSONObjectを継承し、コンストラクタの引数として受け取ったJSONObjectから取得したJSON形式の文字列を
親クラスのコンストラクタに渡すことでメンバでtestResultを持つ必要が無くなりました。

otameshi-developer replied:
うんうん、いいと思います！
このクラスの内容的に、JSONの機能を保持していて全然okだから、継承してしまった方が
役割が明確かな。
※時と場合によるから一概には言えないけど、継承するのかwrapするのかinterfaceを使うのか、
　適宜考えていきましょう。

otameshi-developer commented on PhysicalTestResult/src/study/json/PhysicalTestResult.java:
```diff
@@ -0,0 +1,96 @@
+	 * @return　値となる文字列
+	 * @throws JSONException キーに関連付けられた文字列がない
+	 */
+	public String getString(String key) throws JSONException {
```
JSONExceptionは上位に投げないでください。JSONExceptionはRuntimeExceptionです。RuntimeExceptionをthrowすると、Linterによっては指摘を受けます。で、Runtimeを全く関係のない上位クラスでcatchするとわかりずらいです。（意図的に明示的にそこでcatchをする場合を除いては。）なので、JSON関連のクラス付近で適したエラーハンドリングを行うようにしてください。（ここは考えてみよう）

hkaneko16 replied:
JSONExceptionを上位に投げず、メソッド内でcatchしてスタックトレースを吐くように変更しました。
また、エラーが発生した場合は"error"を返すことで、physicalTestResult.htmlの名前の欄にはerrorと表示されます。

otameshi-developer replied:
なるほど。
今回はこの仕様でokだけど、、、
この仕様ってユーザーさんにとって嬉しいものなのかな？
errorって出ても困っちゃうと思う。
（getInt()の-1も同様です。）
（ひとまずgetInt()の方はresolvedにしちゃうね）

hkaneko16 replied:
ユーザーが明示的にエラーだと分かるようにerrorと出す仕様にしたのですが、
改めて見るとerrorだけだと何のエラーかが分からないように思いました。
「管理者に連絡してください」という文言にすれば親切だと思うのですが、どうでしょうか。

otameshi-developer replied:
うん、そんな考え方で良いと思います！※今回は修正しなくていいよ。
今回着目しているエラーって、まず発生しないと思うのね。
JSONのデータ型が間違っている場合はもちろん発生するけど、それってそもそも変なファイル読み込んでいるのがいけないわけじゃん？
そんな時は値の一部でエラーを表しても伝わらないし、「ちゃんと動いていないよ！」ってのを前面に出す感じで
エラー画面を表示するのがいいんじゃないかな。

> 「管理者に連絡してください」という文言にすれば親切だと思うのですが、どうでしょうか。

表示する文言としてはこれもokだと思うけど、
その前に1文下記のような文言を入れておけば、ユーザーさんも管理者に問い合わせ易くなるんじゃないかな。
「データの読み取りに失敗しました。」
「権限がありません。」※↓ここからは今回のエラーとは関係なく、一般的にありそうな内容。
「ページの有効期限が切れています。」
などなど

otameshi-developer commented on PhysicalTestResult/src/study/json/PhysicalTestResult.java:
```diff
@@ -0,0 +1,96 @@
+	 * @return　値となる数値
+	 * @throws JSONException キーに関連付けられた数値がない
+	 */
+	public int getInt(String key) throws JSONException {
```
getStringのthrowsの指摘と同様

hkaneko16 replied:
こちらもメソッド内でJSONExceptionをcatchしスタックトレースを吐くようにしています。
エラーが発生した場合は、-1を返すことで、physicalTestResult.htmlの名前以外の記録の欄には-1と表示され、ソートもされません。

otameshi-developer replied:
getString()と同様。ひとまずresolvedにします。

hkaneko16 committed:
jarファイルのパスを変更

hkaneko16 committed:
base.html:不要コメントの削除

otameshi-developer commented on PhysicalTestResult/.classpath:
```diff
@@ -2,6 +2,6 @@
 	<classpathentry kind="src" path="src"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="lib" path="C:/Users/Hiroto/Downloads/json-20190722.jar"/>
+	<classpathentry kind="lib" path="C:/Downloads/json-20190722.jar"/>
```
※まぁ直さなくてもいいんですが。。。
個人のPCなのでこの辺はあまり色々言いませんが、
C直下に「Downloads」で、そこに参照jarが入っている構成ってあまりみないよね。。。

こういう時のフォルダ名って大体「lib」とか「libs」かな。（libraryの略）
Downloadsって、Windowsのデフォルトで存在するフォルダだし、
今後、"ダウンロードわけではない"jarを参照したりする場合もあるよね？
その時には「Downloads」には入れられないけど、「lib」とかなら違和感ない。

こんな感じで今後の発展とかも考えて構成や実装していきたいね。

otameshi-developer replied:
直すなら、直す。
直さないなら、この会話resolvedにしてください。
今回はどちらでも構いません。

hkaneko16 replied:
libフォルダ配下に格納しました。

otameshi-developer replied:
確認しました。okです。

otameshi-developer replied:
確認しました。okです・

hkaneko16 committed:
jarファイルをlibフォルダに格納

hkaneko16 committed:
Main.javaの指摘修正

hkaneko16 committed:
FileReader.javaの指摘修正

hkaneko16 committed:
FileReader.java:BufferedReaderから読み込まれるように修正

hkaneko16 committed:
PhysicalTestResult.java:JSONObjectを継承した子クラスに変更
メンバ変数のtestResultを削除し、コンストラクタで受け取った引数は親クラスのコンストラクタに渡すよう修正
JSONExceptionは上位に投げず自クラスで解決するよう修正
不要コメントを削除
変数名を変更

otameshi-developer commented on PhysicalTestResult/src/study/json/PhysicalTestResult.java:
```diff
@@ -5,91 +5,81 @@
+			return -1;
+		}
 
 	}
```
なかなかいい感じになってきたと思います。
、、、が、
throws宣言とかreturnの内容が変わっているけど、これを読んでいる上位クラスには影響はないの？
（ソース修正はない？）
あるのであれば、一緒にコミットすべきかなと。
ちなみに、errorとか-1が返る時はユーザー的にはどんな動きになるの？

otameshi-developer replied:
> ちなみに、errorとか-1が返る時はユーザー的にはどんな動きになるの？

元の会話の所に書いてあったね！
ひとまず了解しました！
※けどエラーハンドリングは変わると思うので、その辺は一緒に修正してね。
　コミットする際はコンパイルエラーがないことが大前提です。
　過不足なく、修正内容をコミットできるようになりましょう。

hkaneko16 replied:
上位クラスのソース修正はあります。
ですが、修正中のため別々にコミットする判断をしました。
影響がある場合は一緒にコミットする → 承知しました。

hkaneko16 committed:
HtmlCreator.java:HTMLBuilderからHtmlCreatorにクラス名変更
繰り返し処理をIteratorではなくforEachで行うように変更
変数名の変更
不要コメントの削除
JSONExceptionのthrows宣言を削除
